server/src/repositories/asset.repository.ts

```diff
@@ -709,4 +713,27 @@ export class AssetRepository implements IAssetRepository {
   async upsertFiles(files: { assetId: string; type: AssetFileType; path: string }[]): Promise<void> {
     await this.fileRepository.upsert(files, { conflictPaths: ['assetId', 'type'] });
   }
+
+  updateOffline(importPaths: string[], exclusionPatterns: string[]): Promise<UpdateResult> {
+    const paths = importPaths.map((importPath) => `${importPath}%`).join('|');
+    const exclusions = exclusionPatterns.map((pattern) => globToSqlPattern(pattern)).join('|');
+    return this.repository
+      .createQueryBuilder()
+      .update()
+      .set({
+        isOffline: true,
+        deletedAt: new Date(),
+      })
+      .where({ isOffline: false })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where('originalPath NOT SIMILAR TO :paths', {
```

Comment:
Use `LIKE` instead of `SIMILAR TO`. 

The exclusions and import paths are also specific to a particular library, right? So you need to specify the library in the query.

Also, can you generate SQL for this and confirm with `EXPLAIN ANALYZE` that it uses an index?

Reply:
Added the library parameter, thanks for spotting.

What is the rationale for using LIKE instead of SIMILAR TO?

Reply:
The Postgres devs [rather dislike it](https://dba.stackexchange.com/a/10696). Queries that can use LIKE should use LIKE, queries that need regex should use the regex operator.